### DISCUSSION TITLE
Bringing together results on ddp on a single machine

### DISCUSSION THREAD

**Discussion by Arij-Aladel** [Lightning-AI/pytorch-lightning#5886] [DDP / multi-GPU / multi-node]
I have the same problem as mentioned in    [1](https://github.com/PyTorchLightning/pytorch-lightning/issues/4175#issue-722428017)  and  [2](https://github.com/PyTorchLightning/pytorch-lightning/issues/702),  the solution was to use dist.all_gather() inside validation_epoch_end . The difference is that I have as output in each validation_step a dictionary:

{"val_loss" : float_loss,  "batch_length": int_len, "preds": text, "answer":text, "doc": `text}.`

Is there away to collect text results from all processes?

I can collect loss using dist.all_gather(), but what about text results any suggestions?
